src/crafting.cpp

```diff
@@ -1293,6 +1307,20 @@ void pick_recipes(const inventory &crafting_inv,
                         continue;
                     }
                 }
+                if(search_qualities) {
+                  bool match_found = false;
+                  itype *it = item::find_type(rec->result);
+                  for(std::map<std::string, int>::const_iterator quality = it->qualities.begin(); quality != it->qualities.end() && !match_found; ++quality) {
```

Comment:
^ Equivalent range-based loop plus `if(match_found) { break; }`.

Also, iterators are why `auto` was invented, I'm pretty sure (only slightly tongue-in-cheek).